src/targets/maven.ts

```diff
@@ -258,12 +261,92 @@ export class MavenTarget extends BaseTarget {
    * @param distDir directory of the distribution.
    */
   private async uploadDistribution(distDir: string): Promise<void> {
+    const bomFile = await this.getBomFileFomDist(distDir);
+    if (bomFile) {
+      this.logger.debug('Found BOM: ', bomFile);
+      await this.uploadBomDistribution(bomFile);
+    } else {
+      this.logger.debug('Did not find a BOM.');
+      await this.uploadPomDistribution(distDir);
+    }
+  }
+
+  private async getBomFileFomDist(
+    distDir: string
+  ): Promise<string | undefined> {
+    const pomFilepath = join(distDir, POM_DEFAULT_FILENAME);
+    if (await this.isBomFile(pomFilepath)) {
+      return pomFilepath;
+    }
+
+    // There may be several files in the ZIP-ed artifact with the same name,
+    // where the BOM may be one of them (there may not be a BOM). Files may be
+    // renamed when extracting the ZIP, so the default name (`pom-default.xml`)
+    // may not match. It's assumed that any renaming keeps the same extension,
+    // so all files with the same extension are checked to identify the BOM.
+    const filesInDir = await fsPromises.readdir(distDir);
+    const potentialPoms = filesInDir
+      .filter(f => extname(f) === POM_FILE_EXTNAME)
+      .filter(f => f !== POM_DEFAULT_FILENAME)
+      .map(f => join(distDir, f));
+
+    for (const f of potentialPoms) {
+      if (await this.isBomFile(f)) {
+        return f;
+      }
+    }
+
+    return undefined;
+  }
+
+  /**
+   * Returns whether the given POM is a BOM.
+   *
+   * A BOM file is a POM file with the following key:
+   * `<packaging>pom</packaging>`, usually named as `pom-default.xml`.
+   *
+   * @param pomFilepath path to the POM.
+   * @returns true if the POM is a BOM.
+   */
+  public async isBomFile(pomFilepath: string): Promise<boolean> {
+    try {
+      await fsPromises.access(pomFilepath, fsConstants.R_OK);
+      const fileContents = await fsPromises.readFile(pomFilepath, {
+        encoding: 'utf8',
+      });
+      const matchesRequiredKey = BOM_FILE_KEY_REGEXP.test(fileContents);
+      if (matchesRequiredKey) {
+        return true;
+      }
+      return false;
+    } catch (error) {
+      this.logger.warn(
+        `Error checking whether it's a BOM file: ${pomFilepath}\n`,
+        error
+      );
+      return false;
+    }
+  }
+
+  private async uploadBomDistribution(bomFile: string): Promise<void> {
+    await retrySpawnProcess(this.mavenConfig.mavenCliPath, [
+      'gpg:sign-and-deploy-file',
+      `-Dfile=${bomFile}`,
+      `-DpomFile=${bomFile}`,
```

Comment:
don't need to set `-Dtypes=jar,jar` here?

Reply:
No, see https://github.com/getsentry/sentry-java/blob/bf425d511746782ee51a9ac475fa353997d6f34e/scripts/release.kts#L64-L70.